cubiscan/cubiscan.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2019 Camptocamp SA
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0.en.html)
+
+import socket
+
+from cubiscan.command import get_command_registry
+
+EXCLUDED = ['start', 'end', 'cr', 'lf']
+
+
+class CubiScan(object):
+
+    ip_address = None
+    port = None
+    timeout = None
+
+    def __init__(self, ip_address, port, timeout=30):
+        self.ip_address = ip_address
+        self.port = port
+        self.timeout = timeout
+
+    def _make_request(self, command, param=None):
+        """Send a command to the cubiscan and await repsonse and parse it."""
+        command_string = get_command_registry().build_command_string(
+            command, param
+        )
+        conn = socket.create_connection((self.ip_address, self.port))
+        conn.send(command_string)
+        conn.settimeout(self.timeout)
+        try:
+            data = conn.recv()
+        finally:
+            conn.close()
+        return self._parse_response(data)
+
+    def _parse_response(self, command, data):
+        """Parse the response."""
+        mapping, neg_mapping = get_command_registry().get_response_for(command)
+        used_map = mapping
```

Comment:
shouldn't you deepcopy the dict here?

Reply:
why do you think i should

Reply:
re-reading: you only need a reference, fine :)